### PR TITLE
Deferring reduction split

### PR DESCRIPTION
**Pull request by y-sq**
manual test.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -857,6 +864,12 @@ def _is_static(x):
             and _is_static(reduction_numel_hint)
             and _is_static(numel_hint)
         )
+
+        return 4096, 4096
```
Comment (y-sq) [inline torch/_inductor/ir.py:868]: a) Manually split it to the expected tiles.

File: torch/_inductor/scheduler.py [hunk 2]
```diff
@@ -2845,6 +2854,8 @@ def has_shared_data_after_reordering_loop(
         to be compatibile with node1 if that's more efficient.
         """
 
+        node1, node2 = node2, node1
```
Comment (y-sq) [inline torch/_inductor/scheduler.py:2857]: b) reorders the loop while also considering the other reduction ops?

Comment (y-sq) [reply torch/_inductor/scheduler.py:2857]: With a) and b), the generated kernel looks optimal:

Test case:
```
def test(x):
    y = torch.sum(x)
    z = x / 10.0
    z_t = z.t().contiguous().t()
    return y, z, z_t
```

Outputs:
```
0.107ms         0.201 GB         1883.86GB/s     triton_red_fused_clone_div_sum_0
0.002ms         0.000 GB            6.83GB/s     triton_red_fused_sum_1
SUMMARY (/tmp/torchinductor_shuqiyang/tmpnmzpylwk/nh/cnhqyu44azg4rc6in5vzvqyk3ouuk6dzyhkbaiv3jrifejoxhbag.py)
0.11ms           0.20 GB         1842.66GB/s
```
Kernel: P1670240539
